bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/votable/VoTableParser.scala

```diff
@@ -218,125 +215,152 @@ trait StandardAdapter {
-
-  private val ucac4BadMagnitude      = 20.0
-  private val ucac4BadMagnitudeError =  0.9.some
-
-  // UCAC4 ignores A-mags
-  override def ignoreMagnitudeField(v: FieldId): Boolean =
-    v.id === "amag" || v.id === "e_amag"
-
-  // Magnitudes with value 20 or error over or equal to 0.9 are invalid in UCAC4
-  override def validMagnitude(m: Magnitude): Boolean =
-    super.validMagnitude(m)         &&
-      m.value =/= ucac4BadMagnitude &&
-      m.error.map(math.abs) <= ucac4BadMagnitudeError
-
-  // UCAC4 has a few special cases to map magnitudes, g, r and i refer to the Sloan bands g', r' and i'
-  override def parseBand(id: FieldId, band: String): Option[MagnitudeBand] =
-    (id.id, id.ucd) match {
-      case ("gmag" | "e_gmag", ucd) if ucd.includes(UcdWord("em.opt.r")) => Some(MagnitudeBand._g)
-      case ("rmag" | "e_rmag", ucd) if ucd.includes(UcdWord("em.opt.r")) => Some(MagnitudeBand._r)
-      case ("imag" | "e_imag", ucd) if ucd.includes(UcdWord("em.opt.i")) => Some(MagnitudeBand._i)
-      case _                                                             => super.parseBand(id, band)
-    }
-}
 
-case object PPMXLAdapter extends CatalogAdapter with StandardAdapter {
-  val idField  = FieldId("ppmxl",    VoTableParser.UCD_OBJID)
-  val raField  = FieldId("raj2000",  VoTableParser.UCD_RA)
-  val decField = FieldId("decj2000", VoTableParser.UCD_DEC)
-
-  // PPMXL may contain two representations for bands R and B, represented with ids r1mag/r2mag or b1mag/b2mac
-  // The ids r1mag/r2mag are preferred but if they are absent we should use the alternative values
-  val primaryMagnitudesIds   = List("r1mag", "b1mag")
-  val alternateMagnitudesIds = List("r2mag", "b2mag")
-  val idsMapping             = primaryMagnitudesIds.zip(alternateMagnitudesIds)
-
-  // Convert angular velocity to mas per year as provided by ppmxl
-  override def parseAngularVelocity(ucd: Ucd, v: String): CatalogProblem \/ AngularVelocity =
-    CatalogAdapter.parseDoubleValue(ucd, v).map(AngularVelocity.fromDegreesPerYear)
-
-  override def filterAndDeduplicateMagnitudes(magnitudeFields: List[(FieldId, Magnitude)]): List[Magnitude] = {
-    // Read all magnitudes, including duplicates
-    val magMap1 = (Map.empty[String, Magnitude]/:magnitudeFields) {
-      case (m, (FieldId(i, _), mag)) if validMagnitude(mag) => m + (i -> mag)
-      case (m, _)                                           => m
-    }
-    // Now magMap1 might have double entries for R and B.  Get rid of the alternative if so.
-    val magMap2 = (magMap1/:idsMapping) { case (m, (id1, id2)) =>
-      if (magMap1.contains(id1)) m - id2 else m
-    }
-    magMap2.values.toList
+object CatalogAdapter {
+
+  val magRegex = """(?i)em.(opt|IR)(\.\w)?""".r
+
+  def parseDoubleValue(ucd: Ucd, s: String): CatalogProblem \/ Double =
+    \/.fromTryCatchNonFatal(s.toDouble).leftMap(_ => FieldValueProblem(ucd, s))
+
+  case object UCAC4 extends CatalogAdapter with StandardAdapter {
```

Comment:
These are just moved into the `CatalogAdapter` without change other than to add the `catalog` method.